sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -62,10 +66,18 @@ class ProjectOverviewView extends VerticalLayout{
         Label titleLabel = new Label("Project Overview")
         titleLabel.addStyleName(ValoTheme.LABEL_LARGE)
         setupProjects()
-        HorizontalLayout buttonBar = setupButtonLayout()
+
+        HorizontalLayout projectButtonBar = setupButtonLayout()
+        VerticalLayout projectLayout = new VerticalLayout(projectButtonBar,projectGrid)
+        projectLayout.setMargin(false)
+
+        splitPanel = createSplitLayout(projectLayout,failedQCSamplesView)
+        failedQCSamplesView.visibilityChangeListener.add(this)
```

Comment:
As you have a FunctionalInterface with your listener interface there is no need to make this class implement the listener and add itself to a child. I think this makes the hierarchy of the view very hard to understand. You can replace the `this` with a simple closure thus leveraging the advantage we gain with the functional interface.

Reply:
To make the methods smaller and more readable, you could extract a method here that sets up the split panel and provide it with the project grid. I do not understand the purpose of the `projectLayout` here. Maybe you can elaborate a bit on it.

Reply:
The project layout is required to make the project grid and the buttons one layout such as the failing sample QC grid and the hide button is a layout. This is needed to achieve a consistent look within the view.

Reply:
I updated the code according to your suggestion:)